R/setkey.R

```diff
@@ -219,7 +219,7 @@ forder <- function(x, ..., na.last=TRUE, decreasing=FALSE)
   if (ncol(x) == 0L) stop("Attempting to order a 0-column data.table.")
   if (is.na(decreasing) || !is.logical(decreasing)) stop("'decreasing' must be logical TRUE or FALSE")
   cols = substitute(list(...))[-1L]
-  if (identical(as.character(cols),"NULL") || !length(cols)) return(NULL) # to provide the same output as base::order
+  if (identical(as.character(cols),"NULL") || !length(cols) || (length(cols) == 1L && !nzchar(cols))) return(NULL) # to provide the same output as base::order
```

Comment:
this covers `forder(DT, )` case, which errored otherwise later on and with a weird message